rdkit/Chem/SaltRemover.py

```diff
@@ -73,18 +114,27 @@ def _initPatterns(self):
     if self.defnData:
       from rdkit.six.moves import cStringIO as StringIO
       inF = StringIO(self.defnData)
+      with closing(inF):
+        self.salts = []
+        for line in inF:
+          line = line.strip().split('//')[0]
+          if line:
+            if self.format == Format.Smarts:
+              salt = SmartsMolSupplier.molFromSmartsLine(line)
+            elif self.format == Format.Smiles:
+              salt = Chem.MolFromSmiles(line)
+            if salt is None:
+              raise ValueError(line)
+            self.salts.append(salt)
     else:
-      inF = open(self.defnFilename, 'r')
-    with closing(inF):
-      self.salts = []
-      for line in inF:
-        line = line.strip().split('//')[0]
-        if line:
-          splitL = whitespace.split(line)
-          salt = Chem.MolFromSmarts(splitL[0])
-          if salt is None:
-            raise ValueError(line)
-          self.salts.append(salt)
+      if self.format == Format.Smarts:
+        supplier = SmartsMolSupplier(self.defnFilename)
+        self.salts = supplier.salts
+      elif self.format == Format.Mol:
+        supplier = SDMolSupplier(self.defnFilename)
+        self.salts = [mol for mol in supplier]
+      else:
+        raise ValueError('Unsupported format for supplier.')
```

Comment:
Because I did not need that format in my project... :)
Would definitely make sense. I saw that there is a `SmilesMolSupplier` module.